server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -335,6 +336,16 @@ static <Response> ActionListener<Response> runBefore(ActionListener<Response> de
         return assertOnce(new ActionListenerImplementations.RunBeforeActionListener<>(delegate, runBefore));
     }
 
+    /**
+     * Wraps a given listener and returns a new listener which releases the provided {@code releaseBefore}
+     * resource before the listener is notified via either {@code #onResponse} or {@code #onFailure}.
+     */
+    static <Response> ActionListener<Response> releaseBefore(ActionListener<Response> delegate, Releasable releaseBefore) {
```

Comment:
It's always irked me that these params are in the opposite order from their executioin. I guess it's consistent with `runBefore()`. But do contemplate whether you'd prefer this to be more readable the other way round.